Native tests fail when quarkus.package.type=uber-jar

**Describe the bug**
Using a brand new project, adding the property `quarkus.package.type=uber-jar` to the `application.properties` so to create an uber jar, the native tests fail to start the native image for an IOException. 

```
[INFO] Running org.acme.getting.started.NativeGreetingResourceIT
Executing [/home/jrota/vagrant/bionic64/issues-reproducers/uber-jar-native-tests-fail/target/uber-jar-native-tests-fail-1.0.0-SNAPSHOT-runner, -Dquarkus.http.port=8081, -Dquarkus.http.ssl-port=8444, -Dtest.url=http://localhost:8081, -Dquarkus.log.file.path=target/target/quarkus.log]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.112 s <<< FAILURE! - in org.acme.getting.started.NativeGreetingResourceIT
[ERROR] org.acme.getting.started.NativeGreetingResourceIT  Time elapsed: 0.111 s  <<< ERROR!
org.junit.platform.commons.JUnitException: Quarkus native image start failed, original cause: java.io.IOException: Cannot run program "/home/jrota/vagrant/bionic64/issues-reproducers/uber-jar-native-tests-fail/target/uber-jar-native-tests-fail-1.0.0-SNAPSHOT-runner": error=2, No such file or directory
Caused by: java.io.IOException: Cannot run program "/home/jrota/vagrant/bionic64/issues-reproducers/uber-jar-native-tests-fail/target/uber-jar-native-tests-fail-1.0.0-SNAPSHOT-runner": error=2, No such file or directory
Caused by: java.io.IOException: error=2, No such file or directory

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   NativeGreetingResourceIT » JUnit Quarkus native image start failed, original c...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```
The only workaround I found was to create a new default profile to create the uber jar so that the native profile stays clean (see https://github.com/kiegroup/kogito-apps/pull/597/commits/950c7f1123c086889a176c623fc1052092463d7b ). 

**Expected behavior**
The native tests should not fail? 

**Actual behavior**
I guess that the property
```
<properties>
        <quarkus.package.type>native</quarkus.package.type>
</properties>
```
is somehow ignored and the native tests try to use the uber jar as well? 

**To Reproduce**
```
mvn io.quarkus:quarkus-maven-plugin:1.10.5.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=uber-jar-native-tests-fail \
    -DclassName="org.acme.getting.started.GreetingResource" \
    -Dpath="/hello"
cd uber-jar-native-tests-fail
mvn clean verify <- no native tests are executed, ok
mvn clean verify -Pnative <- native tests are executed, ok
```
Add `quarkus.package.type=uber-jar` to `application.properties` and then 
```
mvn clean verify -Pnative <- crash
```
See the exception above

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

https://github.com/r00ta/issues-reproducers/tree/main/uber-jar-native-tests-fail

Or attach an archive containing the reproducer to the issue.

Steps to reproduce the behavior:
1. Create a new application
2. Add `quarkus.package.type=uber-jar` to `application.properties`
3. run `mvn clean verify -Pnative`

**Configuration**
```properties
quarkus.package.type=uber-jar
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux jrota.remote.csb 4.18.0-80.11.2.el8_0.x86_64 #1 SMP Sun Sep 15 11:24:21 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: 
 ```
 openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: `1.10.5.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): `mvn`

**Additional context**
(Add any other context about the problem here.)

--- a/devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java
@@ -8,6 +8,7 @@ import java.nio.file.Path;
 import java.util.List;
 
 import org.apache.maven.artifact.Artifact;
+import org.apache.maven.model.Profile;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.logging.Log;
 import org.apache.maven.plugins.annotations.Component;
@@ -30,6 +31,9 @@ import io.quarkus.bootstrap.util.IoUtils;
 public class BuildMojo extends QuarkusBootstrapMojo {
 
     public static final String QUARKUS_PACKAGE_UBER_JAR = "quarkus.package.uber-jar";
+    private static final String PACKAGE_TYPE_PROP = "quarkus.package.type";
+    private static final String NATIVE_PROFILE_NAME = "native";
+    private static final String NATIVE_PACKAGE_TYPE = "native";
 
     @Component
     private MavenProjectHelper projectHelper;
@@ -78,6 +82,20 @@ public class BuildMojo extends QuarkusBootstrapMojo {
     protected void doExecute() throws MojoExecutionException {
 
         try {
+            boolean clearPackageTypeSysProp = false;
+            // Essentially what this does is to enable the native package type even if a different package type is set
+            // in application properties. This is done to preserve what users expect to happen when
+            // they execute "mvn package -Dnative" even if quarkus.package.type has been set in application.properties
+            if (!System.getProperties().containsKey(PACKAGE_TYPE_PROP)
+                    && isNativeProfileEnabled()) {
+                Object packageTypeProp = mavenProject().getProperties().get(PACKAGE_TYPE_PROP);
+                String packageType = NATIVE_PACKAGE_TYPE;
+                if (packageTypeProp != null) {
+                    packageType = packageTypeProp.toString();
+                }
+                System.setProperty(PACKAGE_TYPE_PROP, packageType);
+                clearPackageTypeSysProp = true;
+            }
             try (CuratedApplication curatedApplication = bootstrapApplication()) {
 
                 AugmentAction action = curatedApplication.createAugmentor();
@@ -106,10 +124,26 @@ public class BuildMojo extends QuarkusBootstrapMojo {
                                 result.getJar().getClassifier());
                     }
                 }
+            } finally {
+                if (clearPackageTypeSysProp) {
+                    System.clearProperty(PACKAGE_TYPE_PROP);
+                }
             }
         } catch (Exception e) {
             throw new MojoExecutionException("Failed to build quarkus application", e);
         }
+    }
+
+    private boolean isNativeProfileEnabled() {
+        List<Profile> activeProfiles = mavenProject().getActiveProfiles();
+        if (activeProfiles != null) {
+            for (Profile activeProfile : activeProfiles) {
+                if (NATIVE_PROFILE_NAME.equalsIgnoreCase(activeProfile.getId())) {
+                    return true;
+                }
+            }
+        }
+        return false;
     }
 
     @Override

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/NativeImageIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/NativeImageIT.java
@@ -38,7 +38,7 @@ public class NativeImageIT extends MojoTestBase {
         // trigger mvn package -Pnative -Dquarkus.ssl.native=true
         final String[] mvnArgs = new String[] { "package", "-DskipTests", "-Pnative", "-Dquarkus.ssl.native=true" };
         final MavenProcessInvocationResult result = running.execute(Arrays.asList(mvnArgs), Collections.emptyMap());
-        await().atMost(5, TimeUnit.MINUTES).until(() -> result.getProcess() != null && !result.getProcess().isAlive());
+        await().atMost(10, TimeUnit.MINUTES).until(() -> result.getProcess() != null && !result.getProcess().isAlive());
         final String processLog = running.log();
         try {
             assertThat(processLog).containsIgnoringCase("BUILD SUCCESS");

--- a/integration-tests/spring-data-jpa/src/main/resources/application.properties
+++ b/integration-tests/spring-data-jpa/src/main/resources/application.properties
@@ -1,3 +1,4 @@
+quarkus.package.type=fast-jar
 quarkus.datasource.db-kind=h2
 quarkus.datasource.jdbc.url=jdbc:h2:tcp://localhost/mem:test
 quarkus.datasource.jdbc.max-size=8

